test/cuttlefish_escript_test.erl

```diff
@@ -98,6 +98,35 @@ describe_prints_not_configured() ->
                    ?assertPrinted(Text)
                end).
 
+
+get_test_() ->
+  [
+    {"`cuttlefish get` prints value", fun get_prints/0},
+    {"`cuttlefish get` prints datatype's valid values", fun get_prints_nothing/0}
+  ].
+
+get_(Key) ->
+  ?assertThrow(stop_deactivate, cuttlefish_escript:main(["-i", tp("riak.schema"), "-c", tp("riak.conf"), "get", Key])).
+
+get_prints() ->
+  ?capturing(begin
+               get_("ring_size"),
+               {ok, Stdout0} = cuttlefish_test_group_leader:get_output(),
+               ?assertEqual([["32", $\n]], Stdout0)
+             end),
+  ?capturing(begin
```

Comment:
Unrelated to the code. Just a thought: `capturing` macro could be changed to 

```
define(capturing(__Forms),
       (fun() ->      
            ___OldLeader = group_leader(),
            group_leader(cuttlefish_test_group_leader:new_group_leader(self()), self()),
            try
                __Forms
            after
                cuttlefish_test_group_leader:tidy_up(___OldLeader)
            end
        end)()).
```
To add lexical scope to the variable bindings. This will remove the necessity to introduce `Stdout0`, `Stdout1`, etc.

Reply:
I am not familiar with it. Could you show me the suggested changes on `Stdout0` etc?

Reply:
This macro is defined in the beginning of this module. Currently it wraps code in `begin ... end`, which doesn't prevent variable bindings from escaping the scope. Replacing `begin .. end` with `(fun() -> .. end)()` prevents that. It means `get_prints` function could be rewritten as

```
get_prints() ->
  ?capturing(begin
               get_("ring_size"),
               {ok, Stdout} = cuttlefish_test_group_leader:get_output(),
               ?assertEqual([["32", $\n]], Stdout)
             end),
  ?capturing(begin
               get_("listener.http.internal"),
               {ok, Stdout} = cuttlefish_test_group_leader:get_output(),
               ?assertEqual([["127.0.0.1:8098", $\n]], Stdout)
             end).
```
It's a little less confusing, imho

Reply:
now I get it. nice idea